### PR TITLE
App dropdown fields should have a searchbox

### PR DESCRIPTION
**Pull request by user-yormen**
#1633

### CODE REVIEW

File: sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlSearchableSpinnerField.java [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+import java.util.List;
+
+import de.symeda.sormas.app.R;
+import de.symeda.sormas.app.backend.config.ConfigProvider;
+import de.symeda.sormas.app.component.Item;
+import de.symeda.sormas.app.component.VisualState;
+import de.symeda.sormas.app.component.VisualStateControlType;
+import de.symeda.sormas.app.util.DataUtils;
+
+public class ControlSearchableSpinnerField extends ControlPropertyEditField {
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlSearchableSpinnerField.java:47]: I would prefer this class to extend ControlSpinnerField instead and override the methods that actually have changes

File: sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlSearchableSpinnerFieldListeners.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+import android.view.View;
+import android.widget.AdapterView;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ControlSearchableSpinnerFieldListeners implements AdapterView.OnItemSelectedListener {
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlSearchableSpinnerFieldListeners.java:29]: Remove this class and use ControlSpinnerFieldListeners instead (which should work when ControlSearchableSpinnerField extends ControlSpinnerField)

File: sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlTextEditFieldWithClear.java [hunk 3]
```diff
@@ -0,0 +1,515 @@
...
+import androidx.databinding.InverseBindingAdapter;
+import androidx.databinding.InverseBindingListener;
+
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.app.R;
+import de.symeda.sormas.app.backend.config.ConfigProvider;
+import de.symeda.sormas.app.component.VisualState;
+import de.symeda.sormas.app.component.VisualStateControlType;
+
+public class ControlTextEditFieldWithClear extends ControlPropertyEditField<String> {
```
Comment (MateStrysewskeSym) [inline sormas-app/app/src/main/java/de/symeda/sormas/app/component/controls/ControlTextEditFieldWithClear.java:50]: Same as with the searchable spinner field; make this class extend ControlTextEditField instead

**Comment by MateStrysewskeSym** [comment]
One thing I've noticed: It would be good if the first entry in the filtered list would not be displayed bold and italic; that should only be the case if "Clear" is the first entry.

**Comment by MateStrysewske** [comment]
Closing this pull request because there's been no updates for more than three months. Please submit a new pull request if you're still interested in getting these changes into the software.

**Pull request closed without merging**